domain/schema/model/sql/0020-unit.sql

```diff
@@ -321,3 +321,25 @@ CREATE TABLE resolve_mode (
 INSERT INTO resolve_mode VALUES
 (0, 'retry-hooks'),
 (1, 'no-hooks');
+
+CREATE VIEW v_unit_attribute AS
+SELECT
+    u.uuid,
+    u.name,
+    u.life_id,
+    ur.mode_id AS resolve_mode_id,
+    k.provider_id
+FROM unit AS u
+LEFT JOIN unit_resolved AS ur ON u.uuid = ur.unit_uuid
+LEFT JOIN k8s_pod AS k ON u.uuid = k.unit_uuid;
+
+-- Ensure this is a JOIN not a LEFT JOIN. We don't want to export units
+-- without a machine.
```

Comment:
why?

Reply:
The status will guard against this issue now.

I'll remove it.